docs/snippets/config/trt-llm-params.mdx

```diff
@@ -0,0 +1,148 @@
+### `trt_llm`
+
+Configuration for TRT-LLM accelerated model services. Here's a TinyLlama example to get started quickly:
+
+```yaml
+model_name: TinyLlama-1.1B-Chat TRT-LLM
+resources:
+  accelerator: A10G
+  use_gpu: True
+trt_llm:
+  build:
+    max_input_len: 1000
+    max_batch_size: 1
+    max_beam_width: 1
+    max_output_len: 500
+    strongly_typed: True
+    quantization_type: no_quant
+    base_model: llama
+    checkpoint_repository:
+      repo: TinyLlama/TinyLlama-1.1B-Chat-v1.0
+      source: HF
+```
+#### `build`
+
+TRT-LLM engine build configuration.
+
+##### `base_model`
+
+The base model architecture of your model checkpoint. Supported architectures include:
+
+* `llama`
+* `mistral`
+* `deepseek`
+* `whisper`
+
+##### `max_input_len`
+
+Maximum input length in tokens to limit prompts to.
+
+##### `max_output_len`
+
+Maximum output length in tokens to limit outputs to.
+
+##### `max_batch_size`
+
+Maximum number of input sequences to pass through the engine consecutively.
+
+##### `max_beam_width`
+
+Maximum number of candidate sequences with which to conduct beam search.
+
+##### `max_prompt_embedding_table_size`
+(default: `0`)
+
+Maximum prompt embedding table size for prompt tuning.
+
+##### `checkpoint_repository`
+
+Specification of the model checkpoint to be leveraged for engine building. E.g.
+
+```yaml
+checkpoint_repository:
+    source: HF | GCS | REMOTE_URL
+    repo: meta-llama/Meta-Llama-3-8B-Instruct | gs://bucket_name | https://your-checkpoint.com
+```
+
+To configure access to private model checkpoints, [register secrets in your Baseten workspace](https://docs.baseten.co/observability/secrets#best-practices-for-secrets), namely the `hf_access_token` or `trt_llm_gcs_service_account` secrets with a valid service account json for HuggingFace or GCS, respectively.
+Ensure that you push your truss with the `--trusted` flag to enable access to your secrets.
+
+###### `checkpoint_repository.source`
+
+Source where the checkpoint is stored. Supported sources include:
+* `HF` (HuggingFace)
+* `GCS` (Google Cloud Storage)
+* `REMOTE_URL`
+
+###### `checkpoint_repository.repo`
+
+Checkpoint repository name, bucket, or url.
+
+##### `gather_all_token_logits`
+(default: `False`)
+
+Whether to gather all token logits.
+
+##### `strongly_typed`
+(default: `False`)
+
+Whether to build the engine using strong typing, can speed up build time for some formats.
```

Comment:
can we ad a note on when this is useful to set to True (or link to docs)? I don't have a good sense